### ISSUE TITLE
New string constants are problematic

### ISSUE THREAD

**Issue by kennykerr** [microsoft/win32metadata#783]
A bunch of new string literals started showing up. Some of them are a little tricky to parse. Wondering whether they are correct and whether they are valid. For example:

```C++
#define GUID_NTDS_QUOTAS_CONTAINER_BYTE "\x62\x27\xf0\xaf\x1f\xc2\x41\x0d\x8e\x3b\xb1\x06\x15\xbb\x5b\x0f"
```

Ends up as:

```C#
public const string GUID_NTDS_QUOTAS_CONTAINER_BYTE = "b'ð\u00af\u001fÂA\r\u008e;±\u0006\u0015»[\u000f";
```

I'm not sure that's meant to be treated as a string at all, but I don't really know. Either way, I'm finding some compilers have a hard time with such string literals. 

I suspect it is meant to be treated as a byte array and not a Unicode string.

**Comment by kennykerr** [comment]
Note also that the actual bytes written to the winmd are different. So, a parser can't even ignore the encoding and just assume the bytes are the same:

```
Before: \x62\x27\xf0\xaf\x1f\xc2\x41\x0d\x8e\x3b\xb1\x06\x15\xbb\x5b\x0f
After:  \x62\x27\xc3\xb0\xc2\xaf\x1f\xc3\x82\x41\x0d\xc2\x8e\x3b\xc2\xb1\x06\x15\xc2\xbb\x5b\x0f
```

**Comment by sotteson1** [comment]
It does look like those are meant to be byte arrays in the header. Are you OK if I just exclude them? We don't currently have a way to expose byte arrays as constants.

**Comment by kennykerr** [comment]
Yep, that's fine by me. We can always reevaluate if there's customer demand.

**Comment by riverar** [comment]
I have no objection to dropping these _fake_ string constants for now, however there appears to be a deeper issue here where ANSI string literal constants are being treated as wide strings and ultimately getting encoded into the winmd as such.

Example:
```
#define GUID_NTDS_QUOTAS_CONTAINER_BYTE "\x62\x27\xf0\xaf\x1f\xc2\x41\x0d\x8e\x3b\xb1\x06\x15\xbb\x5b\x0f"
```

is getting stored in metadata as

```
.field public static literal string GUID_NTDS_QUOTAS_CONTAINER_BYTE
  = bytearray (62 00 27 00 F0 00 AF 00 1F 00 C2 00 41 00 0D 00
               8E 00 3B 00 B1 00 06 00 15 00 BB 00 5B 00 0F 00)
```

but should really be

```
.field public static literal string GUID_NTDS_QUOTAS_CONTAINER_BYTE
  = bytearray (62 27 F0 AF 1F C2 41 0D 8E 3B B1 06 15 BB 5B 0F ..)
```

(ECMA-335 doesn't dictate how Unicode strings are encoded and stored in the string heap but it appears ildasm/ilasm assume 1 or 2 byte UTF-8 encoding. So I expect ANSI strings to always be represented with a vector of 7-bit bytes.)

**Comment by riverar** [comment]
(cont) Another example:

Constants

```
#define GUID_COMPUTRS_CONTAINER_A                   "aa312825768811d1aded00c04fd8d5cd"
#define GUID_COMPUTRS_CONTAINER_W                   L"aa312825768811d1aded00c04fd8d5cd"
```

are getting emitted as

```
.field public static literal string GUID_COMPUTRS_CONTAINER_A
  = bytearray ( 61 00 61 00 33 00 31 00 32 00 38 00 32 00 35 00   // a.a.3.1.2.8.2.5.
                37 00 36 00 38 00 38 00 31 00 31 00 64 00 31 00   // 7.6.8.8.1.1.d.1.
                61 00 64 00 65 00 64 00 30 00 30 00 63 00 30 00   // a.d.e.d.0.0.c.0.
                34 00 66 00 64 00 38 00 64 00 35 00 63 00 64 00 ) // 4.f.d.8.d.5.c.d.

.field public static literal string GUID_COMPUTRS_CONTAINER_W
  = bytearray ( 61 00 61 00 33 00 31 00 32 00 38 00 32 00 35 00   // a.a.3.1.2.8.2.5.
                37 00 36 00 38 00 38 00 31 00 31 00 64 00 31 00   // 7.6.8.8.1.1.d.1.
                61 00 64 00 65 00 64 00 30 00 30 00 63 00 30 00   // a.d.e.d.0.0.c.0.
                34 00 66 00 64 00 38 00 64 00 35 00 63 00 64 00 ) // 4.f.d.8.d.5.c.d.
```

But perhaps we can open a new issue for this.

**Comment by kennykerr** [comment]
I just tried 17.0.5 and there are more problematic string literals:

* GUID_PRIVILEGED_ACCESS_MANAGEMENT_OPTIONAL_FEATURE_BYTE
* GUID_RECYCLE_BIN_OPTIONAL_FEATURE_BYTE
